examples/dataflow-cloud-run/dataflowApp/src/test/java/com/demo/dataflow/eventfn/ParseGoBikeEventsTest.java

```diff
@@ -0,0 +1,52 @@
+package com.demo.dataflow.eventfn;
+
+import com.demo.dataflow.model.FailedMessage;
+import com.demo.dataflow.model.GoBike;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class ParseGoBikeEventsTest  implements Serializable {
```

Comment:
Add few more negative test cases like
- null in csv file
- corrupt data
- mix of valid and invalid records


- 

-

Reply:
Hi @ajayydv Added the case of invalid records. I am not sure how to test for CSV file without using resources (such as bucket and BQ table). Did you mean Integration Test?